docs/porting/functionality-differences.md

```diff
@@ -0,0 +1,1051 @@
+  <p><span>RAND_load_file</span></p>
+  </td>
+  <td>
+  <p><span>Returns a non-negative number.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_write_file</span></p>
+  </td>
+  <td>
+  <p><span>Does nothing and returns negative
+  one.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_file_name</span></p>
+  </td>
+  <td>
+  <p><span>Returns NULL.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_add</span></p>
+  </td>
+  <td>
+  <p><span>Does nothing.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_egd</span></p>
+  </td>
+  <td>
+  <p><span>Returns 255.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_poll</span></p>
+  </td>
+  <td>
+  <p><span>Returns one.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_status</span></p>
+  </td>
+  <td>
+  <p><span>Returns one.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_cleanup</span></p>
+  </td>
+  <td>
+  <p><span>Does nothing.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_SSLeay</span></p>
+  </td>
+  <td>
+  <p><span>Returns a dummy </span>
+  <span>RAND_METHOD</span>
+  <span> pointer.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_OpenSSL</span></p>
+  </td>
+  <td>
+  <p><span>Returns a dummy </span>
+  <span>RAND_METHOD</span>
+  <span> pointer.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_get_rand_method</span></p>
+  </td>
+  <td>
+  <p><span>Returns a dummy </span>
+  <span>RAND_METHOD</span>
+  <span> pointer.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_set_rand_method</span></p>
+  </td>
+  <td>
+  <p><span>Returns one.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>RAND_keep_random_devices_open</span></p>
+  </td>
+  <td>
+  <p><span>Does nothing.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td rowspan=4>
+  <p><span>ASN1</span></p>
+  </td>
+  <td rowspan=4>
+  <p>
+    <span>
+        <a href="https://github.com/aws/aws-lc/blob/5c358103c5df836b9343bf995717b5bc13d5e82f/include/openssl/asn1.h#L1894-L1904">
+            asn1.h
+        </a>
+    </span>
+  </p>
+  </td>
+  <td>
+  <p><span>ASN1_STRING_set_default_mask</span></p>
+  </td>
+  <td>
+  <p><span>Does nothing.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>ASN1_STRING_set_default_mask_asc</span></p>
+  </td>
+  <td>
+  <p><span>Returns one.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>ASN1_STRING_get_default_mask</span></p>
+  </td>
+  <td>
+  <p><span>Returns </span>
+  <span>B_ASN1_UTF8STRING</span>
+  <span> (The default value AWS-LC uses).</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td>
+  <p><span>ASN1_STRING_TABLE_cleanup</span></p>
+  </td>
+  <td>
+  <p><span>Does nothing.</span></p>
+  </td>
+ </tr>
+ <tr>
+  <td rowspan=16>
+  <p><span>Thread Safety</span></p>
+  </td>
+  <td rowspan=16>
+  <p>
+    <span>
+        <a href="https://github.com/aws/aws-lc/blob/5c358103c5df836b9343bf995717b5bc13d5e82f/API-CONVENTIONS.md#thread-safety">
+            Thread Safety
```

Comment:
Is there no header file to be mentioned here where the following values are defined?

Reply:
Sure, I thought this page had more details, but the introduction at the top of [this section](https://github.com/aws/aws-lc/blob/f31f7091ec6b13163c7764519a6d95daefacd6e5/include/openssl/thread.h#L119-L199) explains things well too. Changed.